cfgov/unprocessed/apps/youth-employment-success/js/views/review/choice.js

```diff
@@ -0,0 +1,118 @@
+import { checkDom, setInitFlag } from '../../../../../js/modules/util/atomic-helpers';
+import { updateRouteChoiceAction } from '../../reducers/choice-reducer';
+import { toArray } from '../../util';
+import transportationMap from '../../data/transportation-map';
+import inputView from '../input';
+
+const CLASSES = Object.freeze( {
+  CONTAINER: 'js-yes-review-choice',
+  REVIEW_PLAN: 'js-yes-plans-review',
+  BUTTON: 'js-yes-choice-finalize',
+  CHOICE: 'js-yes-route-selection'
+} );
+
+function reviewChoiceView( element, { store } ) {
+  const _dom = checkDom( element, CLASSES.CONTAINER );
+  const _reviewPlanEl = _dom.querySelector( `.${ CLASSES.REVIEW_PLAN }` );
+  const _choiceBtnEls = toArray(
+    _dom.querySelectorAll( `.${ CLASSES.CHOICE }` )
+  );
+  const _reviewBtnEl = _dom.querySelector( `.${ CLASSES.BUTTON }` );
+
+  function _enableChoices() {
+    _choiceBtnEls.forEach( el => el.querySelector( 'input' ).setAttribute( 'disabled', false )
+    );
+  }
+
+  function _disableChoices() {
+    _choiceBtnEls.forEach( el => el.querySelector( 'input' ).setAttribute( 'disabled', 'disabled' )
+    );
+  }
+
+  function _enableReviewButton() {
+    _reviewBtnEl.setAttribute( 'disabled', '' );
+  }
+
+  function _disableReviewButton() {
+    _reviewBtnEl.setAttribute( 'disabled', 'disabled' );
+  }
+
+  function _populateOptionLabels( routes ) {
+    _choiceBtnEls.forEach( ( el, index ) => {
+      const route = routes[index];
+
+      // Need a check here because the third option isnt tied to a route.s
+      if ( route ) {
+        const label = el.querySelector( 'label' );
+        const friendlyOption = transportationMap[route.transportation];
+        const nextLabel = label.textContent.replace( '-', friendlyOption );
+        label.textContent = nextLabel;
+      }
+    } );
+  }
+
+  function _showReviewPlan() {
+    _reviewPlanEl.classList.remove( 'u-hidden' );
+    _reviewBtnEl.removeEventListener( 'click', _showReviewPlan );
+  }
+
+  function _hideReviewPlan() {
+    _reviewPlanEl.classList.add( 'u-hidden' );
```

Comment:
Should you add `_reviewBtnEl.addEventListener( 'click', _showReviewPlan );` here, and call `_hideReviewPlan();` on line 79? Since if you were to call `_showReviewPlan()` and then `_hideReviewPlan()` the event listener added in `_initInputs()` is removed but never re-added.

Reply:
`_hideReviewPlan` will never be called more than once. I just added it as a function because it reads better than `el.classList.etc`